lib/resque/plugins/retry.rb

```diff
@@ -453,6 +453,13 @@ def on_failure_retry(exception, *args)
           return
         end
 
+        # If we are "ignoring" the exception, then we decrement the retry
+        # counter, so that the current attempt didn't count toward the retry
+        # counter.
+        if !@ignore_exceptions.nil? && @ignore_exceptions.include?(exception.class)
```

Comment:
1. I thought we should still want to trigger the try_again and give_up callbacks even after this. Why wouldn't we want to trigger them?

2. Fixed